src/control/destination/background/destination_set.rs

```diff
@@ -302,7 +303,23 @@ fn pb_to_addr_meta(
         }
     };
 
-    let meta = Metadata::new(DstLabels::new(labels.into_iter()), tls_identity);
+    let mut proto_hint = ProtocolHint::Unknown;
+
+    if let Some(hint) = pb.protocol_hint {
+        if let Some(proto) = hint.protocol {
+            match proto {
+                Protocol::H2(..) => {
```

Comment:
tioli i think you could combine this to

```rust
if let Some(Protocol::H2(..)) = hint.protocol {
```

Reply:
Yea, I went back and forth, and ultimately felt being exhaustive on the `Protocol` match was a better idea, since it makes us consider if this needs updating if more hints are added.